TrafficCapture/dockerSolution/src/main/docker/migrationConsole/workflows/documentBulkLoad.yaml

```diff
@@ -0,0 +1,404 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: bulk-load
+  # add labels later w/ patch
+spec:
+  serviceAccountName: argo-workflow-executor
+  entrypoint: run-bulk-load
+
+  templates:
+    - name: run-bulk-load-from-config
+      inputs:
+        parameters:
+          # Required parameters (no default names)
+          - name: target-config
+            description: "Target host config object"
+          - name: snapshot-config
+            description: "Details of the snapshot to use (required)"
+          - name: session-name
+            description: "Name of the unique session to disambiguate coordinating indices"
+          - name: optional-configs
+            description: "JSON Key-Value form of all of the optional parameters to run-bulk-load"
+            value: ""
+      steps:
+        - - name: run-bulk-load
+            template: run-bulk-load
+            arguments:
+              parameters:
+                - name: target-host
+                  value: "{{=fromJSON(inputs.parameters['target-config'])['endpoint']}}"
+                - name: target-insecure
+                  value: "{{=fromJSON(inputs.parameters['target-config'])['allow_insecure']}}"
+                - name: target-username
+                  value: "{{=let a=fromJSON(inputs.parameters['target-config'])['basic_auth'];a==nil?'':a['username']}}"
+                - name: target-password
+                  value: "{{=let a=fromJSON(inputs.parameters['target-config'])['basic_auth'];a==nil?'':a['password']}}"
+                - name: snapshot-name
+                  value: "{{=fromJSON(inputs.parameters['snapshot-config'])['snapshot_name']}}"
+                - name: s3-repo-uri
+                  value: "{{=fromJSON(inputs.parameters['snapshot-config'])['s3']['repo_uri'] ?? ''}}"
+                - name: s3-region
+                  value: "{{=fromJSON(inputs.parameters['snapshot-config'])['s3']['aws_region'] ?? ''}}"
+                - name: s3-endpoint
+                  value: "{{=fromJSON(inputs.parameters['snapshot-config'])['s3']['endpoint'] ?? ''}}"
+                - name: session-name
+                  value: "{{inputs.parameters.session-name}}"
+                  # TODO - parse the other stuff out and into the args.
+
+    - name: run-bulk-load
+      inputs:
+        parameters:
+          # Required parameters (no default names)
+          - name: session-name
+            description: "Name of the unique session to disambiguate coordinating indices"
+          - name: target-host
```

Comment:
There seems to be a lot of duplicate declaration for parameters and arguments, are there ways to avoid this pattern?